src/pool.c

```diff
@@ -697,6 +699,57 @@ account_hr(double *avg, const char *address)
     pthread_rwlock_unlock(&rwlock_acc);
 }
 
+uint64_t
+account_wc(const char *address)
+{
+    account_t *account = NULL;
+    uint64_t wc = 0;
+    pthread_rwlock_rdlock(&rwlock_acc);
+    HASH_FIND_STR(accounts, address, account);
+    if (!account)
+        goto bail;
+    wc = (uint64_t)account->worker_count;
+bail:
+    pthread_rwlock_unlock(&rwlock_acc);
+    return wc;
+}
+
+void
+account_rl(char *rig_list_out, char *end_pt, const char *address)
+{
+    char *body = rig_list_out;
+    char *end = end_pt;
+    bool first = true;
+    account_t *account = NULL;
+
+    if (strlen(address) > ADDRESS_MAX)
+        return;
+    pthread_rwlock_rdlock(&rwlock_acc);
+    HASH_FIND_STR(accounts, address, account);
+    if (!account)
+        goto bail;
+
+    client_t *c = (client_t*)gbag_first(bag_clients);
+    while ((c = gbag_next(bag_clients, 0)))
+    {
+        if (strncmp(c->address, address, ADDRESS_MAX) == 0)
+        {
+            if(first)
```

Comment:
You can drop the `first` variable and just do `if (body == rig_list_out)` because we know if it's the first based on the location `body` points to.